### DISCUSSION TITLE
Strongly typed eDSL for SMTLib2

### DISCUSSION THREAD

**Discussion by bruderj15** [tweag/smtlib-backends#74] [Show and tell]
Hello there.

some time ago I was wondering if you guys had an eDSL for SMTLib2 thats hidden behind your curtains.
Do you?

However, I have created an eDSL on top of `smtlib-backends`, focused on a clean API.
Much like [ekmett/ersatz](https://hackage.haskell.org/package/ersatz) does for QSAT.

You may check it out!
Hackage: [hasmtlib](https://hackage.haskell.org/package/hasmtlib)
GitHub: [Hasmtlib](https://github.com/bruderj15/Hasmtlib)

**Comment by facundominguez** [comment]
No secret eDSL over here :)

That's a pretty interesting development to follow. Thanks for sharing it!
